pkgs/applications/misc/actiona/default.nix

```diff
@@ -0,0 +1,109 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cmake,
+  pkg-config,
+  wrapQtAppsHook,
+
+  bluez,
+  libnotify,
+  libXdmcp,
+  libXtst,
+  opencv,
+  qtbase,
+  qtmultimedia,
+  qtscript,
+  qttools,
+  qtx11extras,
+  qtxmlpatterns,
+
+  textToSpeechSupport ? false,
+  qtspeech,
+}:
+
+let
+  # For some reason qtscript wants to use the same version of qtbase as itself
+  # This override makes it think that they are the same version
+  qtscript' = qtscript.overrideAttrs (oldAttrs: {
+    inherit (qtbase) version;
+    postPatch = ''
+      substituteInPlace .qmake.conf \
+          --replace-fail ${oldAttrs.version} ${qtbase.version}
+    '';
+  });
+in
+stdenv.mkDerivation (finalAttrs: {
+  pname = "actiona";
+  version = "3.10.2";
+
+  src = fetchFromGitHub {
+    owner = "Jmgr";
+    repo = "actiona";
+    rev = "v${finalAttrs.version}";
+    hash = "sha256-4RKCNEniBBx0kDwdHVZOqXYeGCsH8g6SfVc8JdDV0hI=";
+    fetchSubmodules = true;
+  };
+
+  patches =
+    [
+      # Set the proper search location for the `.so` files and the translations
+      ./fix-paths.patch
+    ]
+    # Running with TTS support causes the program to freeze for a few seconds every time at startup,
+    # so we hide it behind an optional flag
+    ++ lib.optional (!textToSpeechSupport) ./disable-tts.patch;
```

Comment:
Wouldn't it be better to use lib.optionals instead?

Reply:
I'll look at it, and if it doesn't look too bad, I'll change it.
I understand that lists are to be expected assigned to the `patches` attribute, and actually using lists better shows what's going on. Though, there is no chance that the optional list will have more than 1 member, because the patch is specifically made to handle the entire flag.